### PR TITLE
Add databricks support

### PR DESCRIPTION
**Pull request by rishsriv**
Currently a draft PR. Must check after adding actual data to databricks

### CODE REVIEW

**Comment by rishsriv** [comment]
Finished testing after setting up a Databricks account. Can confirm that this works well!

Note: this uses a hack right now where we change the `db_type` in databricks request for the `get_query` function to `postgres` – to avoid changing the backend code for now.

File: defog/__init__.py [hunk 1]
```diff
@@ -840,7 +847,7 @@ def get_query(
                 "question": question,
                 "api_key": self.api_key,
                 "previous_context": previous_context,
-                "db_type": self.db_type,
+                "db_type": self.db_type if self.db_type != "databricks" else "postgres",
```
Comment (wongjingping) [inline defog/__init__.py:850]: shall we put a note to update this section once we've updated our backend api to allow db_type=databricks?

**Review by wongjingping — APPROVED** [review]
Thanks for putting this together so quickly!

File: defog/__init__.py (hunk 1 above)
Comment (rishsriv) [reply defog/__init__.py:850]: great point, just added on linear!

**Pull request merged**